extensions/pkg/controller/dnsrecord/reconciler.go

```diff
@@ -133,17 +133,22 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 }
 
 func (r *reconciler) reconcile(ctx context.Context, dns *extensionsv1alpha1.DNSRecord, cluster *extensionscontroller.Cluster, operationType gardencorev1beta1.LastOperationType) (reconcile.Result, error) {
-	if err := controllerutils.EnsureFinalizer(ctx, r.reader, r.client, dns, FinalizerName); err != nil {
+	if err := r.statusUpdater.Processing(ctx, dns, operationType, "Reconciling the dnsrecord"); err != nil {
 		return reconcile.Result{}, err
 	}
 
-	if err := r.statusUpdater.Processing(ctx, dns, operationType, "Reconciling the dnsrecord"); err != nil {
+	hadFinalizer := controllerutil.ContainsFinalizer(dns, FinalizerName)
+	if err := controllerutils.EnsureFinalizer(ctx, r.reader, r.client, dns, FinalizerName); err != nil {
 		return reconcile.Result{}, err
 	}
 
 	r.logger.Info("Starting the reconciliation of dnsrecord", "dnsrecord", kutil.ObjectName(dns))
 	if err := r.actuator.Reconcile(ctx, dns, cluster); err != nil {
 		_ = r.statusUpdater.Error(ctx, dns, reconcilerutils.ReconcileErrCauseOrErr(err), operationType, "Error reconciling dnsrecord")
+		if !hadFinalizer {
```

Comment:
This approach will not work if the controller is interrupted in between (e.g. due to OOM kill, eviction, ...). Does it make sense to add an identifier to the `.status` after the record was created successfully? We could use that information when the `DNSRecord` object is deleted.

Reply:
Should we add a new status field or should we use a special condition for that?

Reply:
What do you think about if we maintain `.status.values` that is eventually equal to `.spec.values` if the resource is up-to-date?

Reply:
Relying on `.status.values` will cause problems during migrating to the new dnsrecord controller. If an existing `DNSRecord` has been marked as deleted and has not been reconciled with the new controller, the `.spec.values` field will be empty and the deletion in the infrastructure will be by-passed leaving orphan records on the DNS server.
It is saver to have a marker in the status that creation of the record has always failed so far.
For this purpose I introduced a condition of type `created`.  After the first successful creation it goes to status `True` and will never be changed again. If the creation fails and the condition is not set, it goes to status `False`. If the creation succeeds later, it goes to the final status `True`. Then it is save to delete the `DNSRecord` without action on the infrastructure if the status is `False`.